### PR TITLE
fix: exhaustive switch and export interfaces

### PR DESCRIPTION
**Pull request by jobo322**
@targos is it ok to use this exhaustive switch, it would throw at compile time if all possibilities of ShapeKind are not handled completely.

is it ok, to export the interfaces in this way?

### CODE REVIEW

File: src/util/getShape1D.ts [hunk 1]
```diff
@@ -19,7 +19,9 @@ export function getShape1D(kind: ShapeKind, shapeOptions = {}) {
     case 'pseudoVoigt':
       return new PseudoVoigt(shapeOptions);
     default:
+      // eslint-disable-next-line no-case-declarations
+      const unHandled: never = kind;
       // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      throw new Error(`getShape1D: unknown shape kind: ${kind}`);
+      throw Error(`Unknown distribution ${unHandled}`);
```
Comment (targos) [inline src/util/getShape1D.ts:25]: why did you change these? there are no typescript errors

Comment (jobo322) [reply src/util/getShape1D.ts:25]: The idea is to make exhaustive this switch, so if we add more options in ShapeKind and do not add the respective cases in the switch, the compiler throws us an error. I am moving the error from the runtime to the compiletime

Comment (targos) [reply src/util/getShape1D.ts:25]: okay, so in that case, wrap the code in a block instead of disabling the eslint error: `default: { ... }`

Comment (targos) [reply src/util/getShape1D.ts:25]: you can also try to enable the `switch-exhaustiveness-check` rule from typescript-eslint

File: src/index.ts [hunk 2]
```diff
@@ -7,3 +7,9 @@ export * from './util/getShape2D';
 
 export * from './shapes/1d/Shape1D';
 export * from './shapes/2d/Shape2D';
+
+export type { XYNumber } from './shapes/2d/gaussian2D/Gaussian2D';
+export type { GaussianClassOptions } from './shapes/1d/gaussian/Gaussian';
+export type { LorentzianClassOptions } from './shapes/1d/lorentzian/Lorentzian';
+export type { PseudoVoigtClassOptions } from './shapes/1d/pseudoVoigt/PseudoVoigt';
+export type { Gaussian2DClassOptions } from './shapes/2d/gaussian2D/Gaussian2D';
```
Comment (jobo322) [inline src/index.ts:15]: are these exports of interfaces well?

Comment (targos) [reply src/index.ts:15]: yes

**Review by targos — APPROVED** [review]
